frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretForm.tsx

```diff
@@ -0,0 +1,125 @@
+  status,
+  isSubmitting,
+}) => {
+  const [name, setName] = React.useState('');
+  const [authType, setAuthType] = React.useState(SecretType.dockerconfigjson);
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const stringData = {};
+  const onNameChanged = (event) => {
+    setName(event.target.value);
+    setFieldValue('secretName', event.target.value);
+  };
+
+  const changeAuthenticationType = (type: SecretType) => {
+    setAuthType(type);
+    setFieldValue('type', type);
+  };
+
+  const onDataChanged = (event) => {
+    authType === SecretType.dockerconfigjson
+      ? setFieldValue('formData', _.mapValues({ '.dockerconfigjson': event }, JSON.stringify))
+      : setFieldValue('formData', event);
+    setFieldTouched('formData', true);
+  };
+
+  return (
+    <div className="odc-secret-form">
+      <h1 className="odc-secret-form__title">Create Source Secret</h1>
+      <p className="odc-secret-form__help-text help-block">
+        Source secrets let you authenticate against a Git server.
+      </p>
+      <div className="form-group">
+        <label className="control-label co-required" htmlFor="secret-name">
+          Secret Name
+        </label>
+        <div>
+          <input
+            className="pf-c-form-control"
+            name="secretName"
+            type="text"
+            onChange={onNameChanged}
+            value={name}
+            aria-describedby="secret-name-help"
+            required
+          />
+          <p className="help-block">Unique name of the new secret.</p>
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label" htmlFor="secret-type">
+          Authentication Type
+        </label>
+        <div className="co-create-secret__dropdown">
+          <Dropdown
+            items={authTypes}
+            dropDownClassName="dropdown--full-width"
+            id="dropdown-selectbox"
+            selectedKey={authType}
+            onChange={changeAuthenticationType}
+          />
+        </div>
+      </div>
+      {renderSecretForm(authType, stringData, onDataChanged)}
+      <ButtonBar errorMessage={status && status.submitError} inProgress={isSubmitting}>
+        <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+          <Button
+            type="button"
+            variant={ButtonVariant.link}
+            onClick={handleSubmit}
+            className="odc-pipeline-resource-param__action-btn"
+            aria-label="create"
+          >
+            <CheckIcon />
+          </Button>
+          <Button
+            type="button"
+            className="odc-pipeline-resource-param__action-btn"
+            variant={ButtonVariant.plain}
+            onClick={handleReset}
+            aria-label="close"
+          >
+            <CloseIcon />
+          </Button>
+        </ActionGroup>
+      </ButtonBar>
+    </div>
+  );
+};
+
+export default SecretForm;
```

Comment:
We should be able to re-use the existing Source Secret Form

Reply:
I am re-using the form field from Source secret form. I am not re-using the complete form so that it will not render the ```form``` under the ```form```.

Reply:
You just mounted the AdvanceSection that is the wrapper around this file in a FormikForm when there is a Modal FormikForm as well... I think we need to wrap those together.